internal/service/clouddiscovery/providers_resource_test.go

```diff
@@ -0,0 +1,798 @@
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccProvidersTags(name, "Amazon Web Services",
+					"single", "role_arn", "dynamic",
+					configAccessId, map[string]string{
+						"tag1": "value1",
+						"tag2": "value2",
+					}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckProvidersExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag1", "value1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag2", "value2")),
+			},
+			// Update and Read
+			{
+				Config: testAccProvidersTags(name, "Amazon Web Services",
+					"single", "role_arn", "dynamic",
+					configAccessId, map[string]string{
+						"tag2": "value2changed",
+						"tag3": "value3",
+					}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckProvidersExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag2", "value2changed"),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag3", "value3")),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccCheckProvidersExists(ctx context.Context, resourceName string, v *clouddiscovery.DiscoveryConfig) resource.TestCheckFunc {
+	// Verify the resource exists in the cloud
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		apiRes, _, err := acctest.BloxOneClient.DiscoveryConfigurationAPIV2.
+			ProvidersAPI.
+			Read(ctx, rs.Primary.ID).
+			Execute()
+		if err != nil {
+			return err
+		}
+		if !apiRes.HasResult() {
+			return fmt.Errorf("expected result to be returned: %s", resourceName)
+		}
+		*v = apiRes.GetResult()
+		return nil
+	}
+}
+
+func testAccCheckProvidersDestroy(ctx context.Context, v *clouddiscovery.DiscoveryConfig) resource.TestCheckFunc {
+	// Verify the resource was destroyed
+	return func(state *terraform.State) error {
+		_, httpRes, err := acctest.BloxOneClient.DiscoveryConfigurationAPIV2.
+			ProvidersAPI.
+			Read(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+				// resource was deleted
+				return nil
+			}
+			return err
+		}
+		return errors.New("expected to be deleted")
+	}
+}
+
+func testAccCheckProvidersDisappears(ctx context.Context, v *clouddiscovery.DiscoveryConfig) resource.TestCheckFunc {
+	// Delete the resource externally to verify disappears test
+	return func(state *terraform.State) error {
+		_, err := acctest.BloxOneClient.DiscoveryConfigurationAPIV2.
+			ProvidersAPI.
+			Delete(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+// RandomNumber Function to generate a random 12-digit number for cloud discovery Role ARN
+func randomNumber() string {
```

Comment:
Random Number function has been added to give a Random 12 digit number for AWS role ARN